### PR TITLE
[Core] Allow structured mesh generation to use different mesh size in each direction

### PR DESCRIPTION
**Pull request by jrubiogonzalez**
**📝 Description**
This PR allows to specify different sizes in each direction for the structured mesh generation. Adding more flexibility.

**🆕 Changelog**
Add option to use different size in each direction  
Added test

### CODE REVIEW

File: kratos/processes/structured_mesh_generator_process.cpp [hunk 1]
```diff
@@ -192,14 +215,15 @@ void StructuredMeshGeneratorProcess::GenerateNodes2D(Point const& rMinPoint, Poi
 void StructuredMeshGeneratorProcess::GenerateNodes3D(Point const& rMinPoint, Point const& rMaxPoint)
 {
     GeometryType::CoordinatesArrayType local_element_size = rMaxPoint - rMinPoint;
-    local_element_size /= mNumberOfDivisions;
-    Point local_coordinates = rMinPoint;
+    local_element_size[0] /= mNumberOfDivisions[0];
+    local_element_size[1] /= mNumberOfDivisions[1];
+    local_element_size[2] /= mNumberOfDivisions[2];    Point local_coordinates = rMinPoint;
```
Comment (roigcarlo) [inline kratos/processes/structured_mesh_generator_process.cpp]: new line

**Review by roigcarlo — COMMENTED** [review]
I would leave the name of the parameter as `number_of_divisions` and check if its a integer or an array (example: `"number_of_divisions": [1, 1, 1]` to be in line with variable values and tables.

Aside from that I see no problems.

File: kratos/processes/structured_mesh_generator_process.cpp (hunk 1 above)
Comment (jrubiogonzalez) [reply kratos/processes/structured_mesh_generator_process.cpp]: That was my first draft, but I didn't like the type changes between double and ints for the array.

**Comment by rubenzorrilla** [comment]
> I would leave the name of the parameter as `number_of_divisions` and check if its a integer or an array (example: `"number_of_divisions": [1, 1, 1]` to be in line with variable values and tables.
> 
> Aside from that I see no problems.

On top of @roigcarlo's observation, I'd like to add that this is the approach we follow in similar cases (see for instance the assign scalar/vector processes).

File: kratos/processes/structured_mesh_generator_process.cpp (hunk 1 above)
Comment (jrubiogonzalez) [reply kratos/processes/structured_mesh_generator_process.cpp]: ping @roigcarlo

**Review by rubenzorrilla — APPROVED** [review]
It looks good to me. Thanks @jrubiogonzalez .

**Pull request merged**